js/ui/windowManager.js

```diff
@@ -609,6 +662,20 @@ const WindowManager = new Lang.Class({
             this._unminimizeWindowDone(shellwm, actor);
             this._mapWindowDone(shellwm, actor);
             this._destroyWindowDone(shellwm, actor);
+            this._rotateInCompleted(actor);
+            this._rotateOutCompleted(actor);
+
+            let activeWaitForRotateTimeout = this._rotateInTimeouts.filter(function(timeout) {
+                return timeout == actor._waitingRotateTimeout;
+            });
```

Comment:
This looks unnecessarily complicated. You should always remove `actor._waitingRotateTimeout` if it's set.